Helper/Order.php

```diff
@@ -731,6 +718,208 @@ public function saveUpdateOrder($reference, $boltTraceId = null, $hookType = nul
         }
     }
 
+    /**
+     * @param \Magento\Quote\Model\Quote $quote
+     * @param string                     $transaction
+     * @return OrderModel
+     * @throws BoltException
+     * @throws LocalizedException
+     */
+    public function preAuthCreateOrder($quote, $transaction)
+    {
+        // check if the order has been created in the meanwhile
+        $order = $existingOrder = $this->getExistingOrder($quote->getReservedOrderId());
+        $isStillNeedCreateNewOrder = true;
+        if ($existingOrder) {
+            $isStillNeedCreateNewOrder = $this->updateExistingOrder($existingOrder, $transaction);
+        }
+
+        if ($isStillNeedCreateNewOrder) {
+            $order = $this->processNewOrder($quote, $transaction);
+
+            if (Hook::$fromBolt) {
+                $order->addStatusHistoryComment(
+                    "BOLTPAY INFO :: THIS ORDER WAS CREATED VIA PRE-AUTH WEBHOOK"
+                );
+            }
+            if ($existingOrder) {
+                $order->addStatusHistoryComment(
+                    "BOLTPAY INFO :: The order was recreated."
```

Comment:
Will this message be clear enough to the merchants? These messages are the most visible thing to them, so they should be really clear and easy to understand why it's happening.

Reply:
removed